dashboard/design/survey_summary_design.md

```diff
@@ -0,0 +1,161 @@
+
+
+# Motivation
+- How to reduce the engineering cost to support 100+ JotForm workshop surveys? (Right now I think we have about 5 to 10.)
+- Can we build a generic pipeline to ingest survey submissions, summarize them, and present the result in workshop dashboard for any survey?
+- How much of the current pipeline will still be relevant if we decide to switch to a different survey platform (Qualtrics, SurveyMonkey etc.) in the future?
+
+
+# Options
+We have 3 following options to create and present JotForm survey summaries in workshop dashboard.\
+Specifically, we need to support at least 3 new surveys in 2019, CSF Deep Dive pre & post workshop surveys and CSF Intro survey.
+
+## 1. Update current pipeline to support 2019 surveys
+We will reuse the current pipeline as much as possible to support 3 new CSF surveys in 2019.
+
+This option is based on `local_workshop_daily_survey_report` pipeline. We reuse its data model `WorkshopDailySurvey` and UI view `local_summer_workshop_daily_survey/results.jsx`. We will modify its logic in to retrieve and summarize survey results, which is currently tightly coupled to summer workshop context. This logic lives in `WorkshopSurveyReportController` and `WorkshopSurveyResultsHelper`.
+
+2 approaches:
+- Modify this logic to make it more generic with the risk of introducing regression (the code is complex).
+- Create a new path separate from the existing one, which means adding substantial duplicated code.
+
+**Pros:** Risk is small (if creating a new path). Amount of new code could is not small but manageable. We do just enough to support 3 new surveys in 2019.
+
+**Cons:** Not scalable when we have more surveys. Could become thrown-away work if we decide to move to generic pipeline later.
+
+**Estimated Cost:** 5 ± 1 days.
+
+## 2. Create survey report in JotForm
+JotForm support creating [Visual Report](https://www.jotform.com/help/187-How-to-Create-a-Visual-Report-with-Your-Form-Submissions) ([example](https://www.jotform.com/report/91067837377065)). We can then embed this report as an iframe into workshop dashboard. Authentication and authorization before showing report will be handled by our side.
+
+**Pros:** The cheapest option in term of engineering cost to enable. Survey owners can create report themselves. Report is automatically updated (don't need to download data locally and upload to Tableau).
+
+**Cons:** Can only do basic calculation and visualization (not the same level as Tableau or PowerBI). User can not interact with the report. Cannot pass parameter to personalize the report.
```

Comment:
I would state this even more strongly - this does not support some of our existing views, for example showing facilitators their averages scores across all workshops they've facilitated. We would have to build an individual report per facilitator which is not feasible.

Reply:
Added.